app/src/main/java/com/x8bit/bitwarden/data/platform/util/SpecialCharWithPrecedenceComparator.kt

```diff
@@ -0,0 +1,62 @@
+package com.x8bit.bitwarden.data.platform.util
+
+import java.util.Locale
+
+/**
+ * String [Comparator] where the characters are compared giving precedence to
+ * special characters.
+ */
+object SpecialCharWithPrecedenceComparator : Comparator<String> {
+    override fun compare(str1: String, str2: String): Int {
+        val minLength = minOf(str1.length, str2.length)
+        for (i in 0 until minLength) {
+            val char1 = str1[i]
+            val char2 = str2[i]
+            val compareResult = compareCharsSpecialCharsWithPrecedence(char1, char2)
+            if (compareResult != 0) {
+                return compareResult
+            }
+        }
+        // If all compared chars are the same give precedence to the shorter String.
+        return str1.length - str2.length
+    }
+}
+
+/**
+ * Compare two characters, where a special character is considered with higher precedence over
+ * letters and numbers. If both characters are a letter and they are equal ignoring the case,
+ * give priority to the lowercase instance. If they are both a digit or a non-equal letter
+ * use the default [String.compareTo] converting the chars to the [Locale] specific uppercase
+ * String.
+ */
+private fun compareCharsSpecialCharsWithPrecedence(c1: Char, c2: Char): Int {
+    return when {
+        c1.isLetterOrDigit() && !c2.isLetterOrDigit() -> 1
+        !c1.isLetterOrDigit() && c2.isLetterOrDigit() -> -1
+        c1.isLetter() && c2.isLetter() && c1.equals(other = c2, ignoreCase = true) -> {
```

Comment:
Renames the file and reorganized the order, but from here is the actual change, if its the same letter.